### DISCUSSION TITLE
Update the pinned pnpm version for Volta

### DISCUSSION THREAD

**Discussion by oscard0m** [renovatebot/renovate#23068] [Suggest an Idea]
### What would you like Renovate to be able to do?

Currently, renovate will update the pinned versions of [`yarn`, `node` and `npm`](https://github.com/renovatebot/renovate/blob/main/lib/modules/manager/npm/readme.md) in the volta key. Is it possible to also support updating the pinned `pnpm` version?

Even though `pnpm` for Volta [is experimental](https://docs.volta.sh/advanced/pnpm), I think it's safe to support it.

**Did you already have any implementation ideas?**

I assume it would be a matter of adding an extra check in [`lib/modules/manager/npm/extract/index.ts`](https://github.com/cjpearson/renovate/blob/bcdf8439d61e0358c6c1484c5fc185dd458c9d9b/lib/modules/manager/npm/extract/index.ts#L251-L273):

```diff
if (depType === 'volta') {
   if (depName === 'node') {
     dep.datasource = GithubTagsDatasource.id;
     dep.packageName = 'nodejs/node';
     dep.versioning = nodeVersioning.id;
   } else if (depName === 'yarn') {
     dep.datasource = NpmDatasource.id;
     dep.commitMessageTopic = 'Yarn';
     const major =
       isVersion(dep.currentValue) && api.getMajor(dep.currentValue);
     if (major && major > 1) {
       dep.packageName = '@yarnpkg/cli';
     }
   } else if (depName === 'npm') {
     dep.datasource = NpmDatasource.id;
+  } else if (depName === 'pnpm') {
+    dep.datasource = NpmDatasource.id;
+    dep.commitMessageTopic = 'pnpm';
   } else {
     dep.skipReason = 'unknown-volta';
   }
   if (!isValid(dep.currentValue)) {
     dep.skipReason = 'unspecified-version';
   }
   return dep;
}
```

And updating the corresponding [test](https://github.com/cjpearson/renovate/blob/bcdf8439d61e0358c6c1484c5fc185dd458c9d9b/lib/modules/manager/npm/extract/index.spec.ts#L482-L513) and [snapshot](https://github.com/cjpearson/renovate/blob/main/lib/modules/manager/npm/extract/__snapshots__/index.spec.ts.snap).

I'm happy to open a PR for this once you validate this makes sense.

**Comment by rarkins** [comment]
PR welcome!
